app/Renderers/CalendarRenderer.php

```diff
@@ -906,6 +897,21 @@ protected function addMultidayEvent(EntityEvent $event, string $date, bool $recu
             }
             $this->events[$extraDate][] = $event;
         }
+        $start = [];
+        $end = [];
+        foreach ($this->events as $date => $events) {
```

Comment:
I'm confused why you are looping on $this->events when this function is already called on each event that is added. This creates a n+1 loop for each event displayed, resetting $this->eventEnd on each event drawn on the calendar.